ui/packages/consul-ui/app/serializers/application.js

```diff
@@ -227,36 +227,38 @@ export default class ApplicationSerializer extends Serializer {
     // create a Set and add version with only major.minor : ex-1.24.6 as 1.24
     let versionSet = new Set();
     payload.forEach(function (item) {
-      if (item.Meta && item.Meta['consul-version'] !== '') {
+      if (item.Meta && item.Meta['consul-version'] && item.Meta['consul-version'] !== '') {
         const split = item.Meta['consul-version'].split('.');
         versionSet.add(split[0] + '.' + split[1]);
       }
     });
 
     const versionArray = Array.from(versionSet);
 
-    // Sort the array in descending order using a custom comparison function
-    versionArray.sort((a, b) => {
-      // Split the versions into arrays of numbers
-      const versionA = a.split('.').map((part) => {
-        const number = Number(part);
-        return isNaN(number) ? 0 : number;
-      });
-      const versionB = b.split('.').map((part) => {
-        const number = Number(part);
-        return isNaN(number) ? 0 : number;
-      });
+    if (versionArray.length > 0) {
```

Comment:
I think we can remove this as sort will be a noop if the array is empty

Reply:
@natmegs Yes! , was extra cautious :) , will update.